integration_tests/src/main/python/parquet_test.py

```diff
@@ -465,3 +464,12 @@ def test_spark_32639(std_input_path):
         lambda spark: spark.read.schema(schema_str).parquet(data_path),
         conf=original_parquet_file_reader_conf)
 
+def test_many_column_project(std_input_path):
+    data_path = f"{std_input_path}/issue-2036"
+    with_gpu_session(
```

Comment:
nit: I would prefer if we validate that the output is the same on both the CPU and the GPU.  Just checking that we don't crash is good, but it is better to validate that we also got the correct answer.

Reply:
Could we have a comment then explaining it?

Reply:
it's probably in line with the comment https://github.com/NVIDIA/spark-rapids/issues/2036#issuecomment-810336351, and parts of the serialization are cached. I could not find a way to uncache it.  Instead added a flag to be able to reorder GPU and CPU runs and then we can hit the issue reliably.